dev/README

```diff
@@ -2,6 +2,38 @@ This directory contains information and tools to help develop the
                              Coq system
                        ======================
 
+Beginners guide
+---------------
+
+The general workflow is to first setup a development environment with
+the correct `configure` settings, then hacking on Coq, make-ing, and testing.
+
+First of all, it is recommended to run the configure with
+    ./configure -local -devel
+
+-local allows `make install` to create a local dev install.
+-devel sets up convenient development options, like treating warnings-as-errors
+
+Next, to conveniently test out changes, it is useful to use
+    make bin/coqtop.byte bin/pluginsbyte
+which build bytecode interpreter versions of coqtop.
+
+To access coqc from coqtop.byte, run:
+    coqtop.byte -compile
```

Comment:
Interesting targets:
- `states` builds just enough to run coqtop (including the .vo files which are automatically loaded)
- `byte` for building all the bytecode stuff

I don't know why you would run coqtop.byte over coqtop outside ocamldebug.